src/contexts/products-context.js

```diff
@@ -0,0 +1,29 @@
+import { createContext, useContext, useState, useEffect } from "react";
+import axios from "axios";
+
+const ProductsContext = createContext(null);
+
+const ProductsProvider = ({ children }) => {
+  const [products, setProducts] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const productData = await axios.get("/api/products");
```

Comment:
no need to call the API with the context as all the products API is only gonna be used when you are on the product page therefore you can call it on the products page itself, that way you would also save on performance.

Reply:
and if you want to use it in multiple pages then creating a custom hook for axios would be a good approach

Reply:
I have used context because I am using products API on two pages. So, instead of writing API calls twice, I have created the contexts for products data.